### PR TITLE
Update CI

### LINKED ISSUE #34: CI fails for Python 3.9
## Expected behavior ##

All CI checks pass.

## Actual behavior ##

CI fails for Python 3.9 because it cannot install an appropriate development version of MDAnalysis. This is due to https://github.com/MDAnalysis/cookiecutter-mdakit/issues/110 .

### PR DESCRIPTION
**Pull request by IAlibay**
Fixes #34

Changes made in this Pull Request:

 - Fix CI, dropping 3.9 testing

TODO
 - Update packaging to reflect 3.10+

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: .github/workflows/gh-ci.yaml [hunk 1]
```diff
@@ -23,31 +23,18 @@ defaults:
...
     strategy:
         fail-fast: false
         matrix:
           os: [macOS-latest, ubuntu-latest, windows-latest]
-          python-version: ${{ fromJSON(needs.environment-config.outputs.python-matrix) }}
+          python-version: ["3.10", "3.11", "3.12"]
           mdanalysis-version: ["latest", "develop"]
+          exclude:
+            - python-version: "3.12"
+              mdanalysis-version: "latest"
```
Comment (orbeckst) [inline .github/workflows/gh-ci.yaml]: Possibly add comment here to tell future maintainers how to handle "latest" and highest Python versions.

**Review by orbeckst — APPROVED** [review]
Thanks, LGTM!

Perhaps add a comment for future maintainers.

File: .github/workflows/gh-ci.yaml (hunk 1 above)
Comment (IAlibay) [reply .github/workflows/gh-ci.yaml]: Done

**Comment by IAlibay** [comment]
Thanks for the review @orbeckst!

**Pull request merged**